org.uqbar.project.wollok.lib/src/wollok/lang/WDouble.xtend

```diff
@@ -4,54 +4,55 @@ import org.uqbar.project.wollok.interpreter.WollokInterpreter
 import org.uqbar.project.wollok.interpreter.WollokRuntimeException
 import org.uqbar.project.wollok.interpreter.core.WollokObject
 import org.uqbar.project.wollok.interpreter.nativeobj.NativeMessage
+import java.math.BigDecimal
 
 /**
  * @author jfernandes
  */
 //TODO: has duplicated code with WInteger
-class WDouble extends WNumber<Double> implements Comparable<WDouble> {
+class WDouble extends WNumber<BigDecimal> implements Comparable<WDouble> {
 	
 	new(WollokObject obj, WollokInterpreter interpreter) {
 		super(obj, interpreter)
 	}
 	
-	def abs() { Math.abs(this.wrapped).asWollokObject }
+	def abs() { this.wrapped.abs.asWollokObject }
 
 	@NativeMessage("+")
 	def plus(WollokObject other) { operate(other) [ doPlus(it) ] }
-		def dispatch Number doPlus(Integer w) { wrapped + w }
-		def dispatch Number doPlus(Double w) { wrapped + w }
+		def dispatch Number doPlus(Integer w) { wrapped + new BigDecimal(w) }
+		def dispatch Number doPlus(BigDecimal w) { wrapped + w }
 		//TODO: here it should throw a 100% wollok exception class
 		def dispatch Number doPlus(Object w) { throw new WollokRuntimeException("Cannot add " + w) }
 
 	@NativeMessage("-")
 	def minus(WollokObject other) { operate(other) [ doMinus(it) ] }
-		def dispatch Number doMinus(Integer w) { wrapped - w }
-		def dispatch Number doMinus(Double w) { wrapped - w }
+		def dispatch Number doMinus(Integer w) { wrapped - new BigDecimal(w) }
+		def dispatch Number doMinus(BigDecimal w) { wrapped - w }
 		def dispatch Number doMinus(Object w) { throw new WollokRuntimeException("Cannot substract " + w) }
 
 	@NativeMessage("*")
 	def multiply(WollokObject other) { operate(other) [ doMultiply(it) ] }
-		def dispatch Number doMultiply(Integer w) { wrapped * w }
-		def dispatch Number doMultiply(Double w) { wrapped * w }
+		def dispatch Number doMultiply(Integer w) { wrapped * new BigDecimal(w) }
+		def dispatch Number doMultiply(BigDecimal w) { wrapped * w }
 		def dispatch Number doMultiply(Object w) { throw new WollokRuntimeException("Cannot multiply " + w) }
 
 	@NativeMessage("/")
 	def divide(WollokObject other) { operate(other) [ doDivide(it) ] }
-		def dispatch Number doDivide(Integer w) { wrapped / w }
-		def dispatch Number doDivide(Double w) { wrapped / w }
+		def dispatch Number doDivide(Integer w) { wrapped / new BigDecimal(w) }
+		def dispatch Number doDivide(BigDecimal w) { wrapped / w }
 		def dispatch Number doDivide(Object w) { throw new WollokRuntimeException("Cannot divide " + w) }
 
 	@NativeMessage("**")
 	def raise(WollokObject other) { operate(other) [ doRaise(it) ] }
 		def dispatch Number doRaise(Integer w) { wrapped ** w }
-		def dispatch Number doRaise(Double w) { wrapped ** w }
+		def dispatch Number doRaise(BigDecimal w) { Math.pow( wrapped.doubleValue , w.doubleValue) }
```

Comment:
Because wrapped.pow(w) only acepts an Integer w. So, other test fails like assert.equals(9 ** 0.5,3).
The bes way to do this without an external library is using Math.pow()